pkg/importer/dataStream.go

```diff
@@ -269,6 +328,128 @@ func (d *DataStream) http() (io.ReadCloser, error) {
+
+	//1. create temporary directory if does not exist to which all the data will be extracted
+	if _, err := os.Stat(tmpData.dataDir); os.IsNotExist(err) {
+		err := os.Mkdir(tmpData.dataDir, os.ModeDir|os.ModePerm)
+		if err != nil {
+			glog.Errorf("Failed to create temporary directory")
+			return nil, errors.Wrapf(err, fmt.Sprintf("Failed to create tempdirectory %s", tmpData.dataDir))
+		}
+	}
+
+	//cleanup in case of failure
+	cleanup := true
+	defer func() {
+		if cleanup {
+			tmpData.Cleanup()
+		}
+	}()
+
+	//2. copy image from registry to the temporary location
+	glog.V(1).Infof("using skopeo to copy from registry")
+	err := image.CopyRegistryImage(d.Endpoint, tmpData.dataDir, ContainerDiskImageDir, d.AccessKey, d.SecKey)
+	if err != nil {
+		glog.Errorf("Failed to read data from registry")
+		return nil, errors.Wrapf(err, fmt.Sprintf("Failed ro read from registry"))
+	}
+
+	//3. Search for file in /disk directory - if not found - failure
+	imageFile, err := getImageFileName(imageDir)
+	if err != nil {
+		glog.Errorf("Error getting Image file from imageDirectory")
+		return nil, errors.Wrapf(err, fmt.Sprintf("Cannot locate image file"))
+	}
+
+	// 4. If found - Create a reader that will read this file and attach it to the dataStream
+	tmpData.file, err = os.Open(filepath.Join(imageDir, imageFile))
+	if err != nil {
+		glog.Errorf("Failed to open image file")
+		return nil, errors.Wrapf(err, fmt.Sprintf("Fail to create data stream from image file"))
+	}
+
+	//got this far - do not cleanup
+	cleanup = false
+
+	d.tmpDataPath = filepath.Join(imageDir, imageFile)
+	d.dataCtxt = tmpData
+	glog.V(1).Infof("Sucecssfully found file. VM disk image filename is %s", imageFile)
+
+	return ioutil.NopCloser(bufio.NewReader(tmpData.file)), nil
+}
+
+func getImageFileName(dir string) (string, error) {
+
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		glog.Errorf("image directory does not exist")
+		return "", errors.Errorf(fmt.Sprintf("image directory does not exist "))
+	}
+
+	entries, err := ioutil.ReadDir(dir)
+	if err != nil {
+		glog.Errorf("Error reading directory")
+		return "", errors.Wrapf(err, fmt.Sprintf("image file does not exist in image directory"))
+	}
+
+	if len(entries) == 0 {
+		glog.Errorf("image file does not exist in image directory - directory is empty ")
+		return "", errors.Errorf(fmt.Sprintf("image file does not exist in image directory - directory is empty"))
+	}
+
+	fileinfo := entries[len(entries)-1]
+	if fileinfo.IsDir() {
+		glog.Errorf("image file does not exist in image directory contains another directory ")
+		return "", errors.Errorf(fmt.Sprintf("image file does not exist in image directory"))
+	}
+
+	filename := fileinfo.Name()
+
+	if len(strings.TrimSpace(filename)) == 0 {
+		glog.Errorf("image file does not exist in image directory - file has no name ")
+		return "", errors.Errorf(fmt.Sprintf("image file does not exist in image directory"))
+	}
+
+	glog.V(1).Infof("VM disk image filename is %s", filename)
```

Comment:
I don't think this is a verbosity 1 level message, its more debugging, so a 3 is more appropriate IMO.

Reply:
done